big_tests/tests/pubsub_SUITE.erl

```diff
@@ -1421,7 +1443,44 @@ disable_delivery_test(Config) ->
 
               pubsub_tools:delete_node(Alice, Node, [])
       end).
+%%-----------------------------------------------------------------
+%% pubsub_item_publisher_option
+%%-----------------------------------------------------------------
+
+get_item_with_publisher_option_test(Config) ->
+    escalus:fresh_story(
+      Config,
+      [{alice, 1}, {bob, 1}],
+      fun(Alice, Bob) ->
+              Node = pubsub_node(),
+              pubsub_tools:create_node(Alice, Node, []),
+
+              pubsub_tools:publish(Alice, <<"item1">>, Node, []),
 
+
+              A = pubsub_tools:get_item(Alice, Node, <<"item1">>,
+                                        [{expected_result, [#{id => <<"item1">>,
+                                                              publisher => escalus_client:full_jid(Alice)}]}]),
+              pubsub_tools:delete_node(Alice, Node, [])
+      end).
+
+receive_item_notification_with_publisher_option_test(Config) ->
+    escalus:fresh_story(
+      Config,
+      [{alice, 1}, {bob, 1}],
+      fun(Alice, Bob) ->
+              Node = pubsub_node(),
+              pubsub_tools:create_node(Alice, Node, []),
+
+              pubsub_tools:subscribe(Bob, Node, []),
+              pubsub_tools:publish(Alice, <<"item1">>, Node, []),
+
+
+              pubsub_tools:receive_item_notification(Bob, #{id => <<"item1">>,
```

Comment:
There is no `pubsub_tools:receive_item_notification/2`. Did you commit everything?

Reply:
Ah, it's `pubsub_tools:receive_item_notification/4` isn't it? Does this function check the `publisher` attribute? Would the test pass if you passed sth else as a publisher value?

Reply:
So this function SHOULD check other than `id` attributes, but it doesn't yet ;)